pkg/sql/sqlstats/sslocal/sql_stats_ingestor.go

```diff
@@ -393,19 +416,40 @@ func NewSQLStatsIngester(
 
 // clearSession removes the session from the registry and releases the
 // associated statement buffer.
-func (i *SQLStatsIngester) clearSession(sessionID clusterunique.ID) {
+func (i *SQLStatsIngester) clearSession(ctx context.Context, sessionID clusterunique.ID) {
 	if b, ok := i.statementsBySessionID[sessionID]; ok {
 		delete(i.statementsBySessionID, sessionID)
 		b.release()
 
+		// Release memory for this session's buffered statements.
+		if i.acc != nil {
+			if size, ok := i.stmtSizes[sessionID]; ok {
```

Comment:
Is it ever possible that we had a session in `statementsBySessionID` but don't have the size entry in `stmtSizes`? I would think not.

The same question goes for other spots. What I'm concerned about is whether we can have memory accounting drift somehow, i.e. we'll keep making new allocations with `Grow` calls but they aren't accompanied by the respective `Shrink` calls, and in the end we have accounting leak.